code/game/objects/items/weapons/implants/implantreagent_vr.dm

```diff
@@ -0,0 +1,108 @@
+
+/obj/item/weapon/implant/reagent_generator/New()
+	..()
+	create_reagents(usable_volume)
+
+/obj/item/weapon/implanter/reagent_generator
+	var/implant_type = /obj/item/weapon/implant/reagent_generator
+
+/obj/item/weapon/implanter/reagent_generator/New()
+	..()
+	imp = new implant_type(src)
+	update()
+	return
+
+/obj/item/weapon/implant/reagent_generator/implanted(mob/living/carbon/source)
+	processing_objects += src
+	to_chat(source, "<span class='notice'>You implant [source] with \the [src].</span>")
+	source.verbs |= assigned_proc
+	return 1
+
+/obj/item/weapon/implant/reagent_generator/process()
+	var/before_gen
+	if(imp_in && generated_reagent)
+		before_gen = reagents.total_volume
+		if(reagents.total_volume < reagents.maximum_volume)
+			if(imp_in.nutrition >= gen_cost)
+				do_generation()
+		else
+			return
+	else
+		imp_in.verbs -= assigned_proc
+		return
+
+	if(reagents)
+		if(reagents.total_volume == reagents.maximum_volume * 0.05)
+			to_chat(imp_in, "<span class='notice'>[pick(empty_message)]</span>")
+		else if(reagents.total_volume == reagents.maximum_volume && before_gen < reagents.maximum_volume)
+			to_chat(imp_in, "<span class='warning'>[pick(full_message)]</span>")
+
+/obj/item/weapon/implant/reagent_generator/proc/do_generation()
+	imp_in.nutrition -= gen_cost
+	reagents.add_reagent(generated_reagent, gen_amount)
+
+/mob/living/carbon/human/proc/use_reagent_implant()
+	set name = "Transfer From Reagent Implant"
+	set desc = "Remove reagents from am internal reagent into a container."
+	set category = "Object"
+	set src in view(1)
+
+	do_reagent_implant()
+
+/mob/living/carbon/human/proc/do_reagent_implant()
+	if(!isliving(usr) || !usr.canClick())
+		return
+
+	if(usr.stat == 1 || (usr.restrained()))
+		return
+
+	var/mob/S = src
+	var/mob/U = usr
+	var/self
+	if(S == U)
+		self = 1
+
+	var/mob/living/carbon/human/user = null
+	if(self)
+		user = src
+	else
+		user = usr
+
+	var/obj/item/weapon/reagent_containers/container = user.get_active_hand()
+	if(!container)
+		to_chat(user,"<span class='notice'>You need an open container to do this!</span>")
+		return
+
+
+	var/obj/item/weapon/implant/reagent_generator/rimplant
+	for(var/I in src.contents)
+		if(istype(I, /obj/item/weapon/implant/reagent_generator))
+			rimplant = I
+			break
+	if(rimplant)
+		if(container.reagents.total_volume < container.volume)
+			var/container_name = container.name
+			if(rimplant.reagents.trans_to(container, amount = rimplant.transfer_amount))
+				user.visible_message("<span class='notice'>[usr] [pick(rimplant.emote_descriptor)] into \the [container_name].</span>",
+									"<span class='notice'>You [pick(rimplant.self_emote_descriptor)] some [rimplant.generated_reagent] into \the [container_name].</span>")
+				if(prob(5))
+					src.visible_message("<span class='notice'>[src] [pick(rimplant.random_emote)].</span>") // M-mlem.
+			if(rimplant.reagents.total_volume == rimplant.reagents.maximum_volume * 0.05)
```

Comment:
I recommend using `<=` here in case future changes make it not land exactly on 5%